_pages/dev/performance/core-web-vitals/cumulative-layout-shift.md

```diff
@@ -0,0 +1,257 @@
+
+Ideally the responsive layout should be controlled just with the static HTML and CSS, but not changed with lately-loaded Javascript.
+
+### Why CSS rules are preferred over JavaScript for responsive layout
+[Angular SSR native feature of inlining critical CSS](https://angular.dev/reference/configs/workspace-config#styles-optimization-options) is responsible for ensuring that the CSS is inlined in the `<head>` of the Server-Side rendered HTML and therefore it's applied by the browser immediately when loading the SSR HTML. Thanks to this, the visual layout is correct from the beginning user seeing the page. This is done automatically by Angular SSR and does not require any additional configuration in Spartacus.
+
+### Recent Spartacus improvements to avoid layout shifts
+
+#### Avoid header layout shift
+Before v2211.43, Spartacus default layout of the header and top navigation on desktop viewport was shifted after the lately-loaded Javascript caused the layout to change. This issue is fixed since Spartacus v2211.43, when the feature toggle `unifiedDefaultHeaderSlotsAcrossBreakpoints` is enabled.
+
+**Important**: Additionally, for apps created before v2211.43 you'll need to also change the deprecated `provideConfig(layoutConfig)` to `provideConfigFactory(layoutConfigFactory)` in your `spartacus-features.module.ts`. 
+
+For more on this issue, read the section below [Appendix: Use the same Header layout configuration for all breakpoints](#Use-the-same-Header-layout-configuration-for-all-breakpoints).
+
+### Use the improved product carousel implementation available since v2211.43
+Before v2211.43, Spartacus used the `<cx-carousel>` component to display product carousels. This carousel implementation caused layout shifts when transitioning from Server-Side Rendered (SSR) HTML to Client-Side Rendered (CSR) HTML, as it changed the DOM structure after the JavaScript was loaded and executed on viewport breakpoints other than mobile. This issue is fixed since Spartacus v2211.43, when the feature toggle `productCarouselScrolling` is enabled.
+
+When `productCarouselScrolling` is enabled, the improved carousel implementation `<cx-carousel-scrolling>` is used instead of the old `<cx-carousel>` as a child of the in the `ProductCarouselComponent` and `ProductReferencesComponent`.
+The new carousel implementation, as opposed to the old one, does not change the DOM structure after transitioning from SSR to CSR, so it avoids layout shifts. As an added benefit, it also more mobile-friendly thanks the swipe gestures allowing to continuously scroll the carousel items (instead of the previous need to click on the next/previous buttons to change slides).
```

Comment:
```suggestion
The new carousel implementation, as opposed to the old one, does not change the DOM structure after transitioning from SSR to CSR, so it avoids layout shifts. As an added benefit, it is also more mobile-friendly thanks the swipe gestures allowing to continuously scroll the carousel items (instead of the previous need to click on the next/previous buttons to change slides).
```